### PR TITLE
.ci/generate_bear_requirements.py:  Generates R dependencies

### PR DESCRIPTION
**Pull request by SubbulakshmiRS**
Changes made in generate_bear_requirements.py and .mobal.yaml to
generate R dependencies using Jinja template
(.moban.dt/Description.jj2)

Closes https://github.com/coala/coala-bears/issues/2482

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: .moban.dt/Description.jj2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Package: coala
+Version: 0.12.0.dev
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: needs to come from the variables in `.moban.yaml`

File: .moban.dt/Description.jj2 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Package: coala
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: needs to come from the variables in `.moban.yaml`

File: .moban.dt/Description.jj2 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Version: 0.12.0.dev
+Title: R dependencies of Coala package
+Author: The coala developers
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: needs to come from the variables in `.moban.yaml`

File: .moban.dt/Description.jj2 [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Mainainer: Lasse Schuirmann, Fabian Neuschmidt, Mischa Krxfcger
+Description: Coala is used for linting and fixing code for all languages.
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: needs to come from the variables in `.moban.yaml`

File: .moban.dt/Description.jj2 [hunk 5]
```diff
@@ -0,0 +1,16 @@
+Package: coala
+Version: 0.12.0.dev
+Title: R dependencies of Coala package
+Author: The coala developers
+Mainainer: Lasse Schuirmann, Fabian Neuschmidt, Mischa Krxfcger
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: `Mainainer` is spelling mistake; and the value needs to come from the variables in `.moban.yaml`

File: .moban.dt/Description.jj2 [hunk 6]
```diff
@@ -0,0 +1,16 @@
+Package: coala
+Version: 0.12.0.dev
+Title: R dependencies of Coala package
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: needs to come from the variables in .moban.yaml

File: .moban.yaml [hunk 7]
```diff
@@ -31,6 +31,7 @@ targets:
   - Gemfile: Gemfile.jj2
   - package.json: package.json.jj2
   - bear-requirements.txt: bear-requirements.txt.jj2
+  - Decription: Description.jj2
```
Comment (jayvdb) [inline .moban.yaml]: please `git add Description` and commit it as part of your commit.

**Comment by jayvdb** [comment]
The commands in https://github.com/coala/coala-bears/blob/master/.ci/deps.r.sh can be simplified to install the packages listed in the new `Description` file..

File: .ci/generate_bear_requirements.py [hunk 8]
```diff
@@ -214,6 +224,8 @@ def sort_requirements(req_dict):
     requirements['gem_requirements'] = get_gem_requirements(gem_reqs)
     requirements['npm_requirements'] = get_npm_requirements(npm_reqs)
     requirements['pip_requirements'] = get_pip_requirements(pip_reqs)
+    requirements['r_requirements'] = get_r_requirements(r_reqs)
```
Comment (nemaniarjun) [inline .ci/generate_bear_requirements.py]: @jayvdb Maybe it is better if we keep these alphabetically sorted. The diff will be large and will need careful reviewing but, probably better in the long term.

File: .ci/generate_bear_requirements.py [hunk 9]
```diff
@@ -152,6 +159,9 @@ def get_pip_requirements(requirements):
     inherited_requirements = get_inherited_requirements()
     return _get_requirements(requirements, '~=', inherited_requirements)
 
+def get_r_requirements(requirements):
+    inherited_requirements=get_inherited_requirements()
+    return _get_requirements(requirements, '>=')
```
Comment (anctartica) [inline .ci/generate_bear_requirements.py]: what was the use of inherited_requirements here?

Comment (anctartica) [reply .ci/generate_bear_requirements.py]: remove it 😄

File: .ci/generate_bear_requirements.py [hunk 10]
```diff
@@ -214,6 +224,8 @@ def sort_requirements(req_dict):
     requirements['gem_requirements'] = get_gem_requirements(gem_reqs)
     requirements['npm_requirements'] = get_npm_requirements(npm_reqs)
     requirements['pip_requirements'] = get_pip_requirements(pip_reqs)
+    requirements['r_requirements'] = get_r_requirements(r_reqs)
+
```
Comment (anctartica) [inline .ci/generate_bear_requirements.py]: remove this blank line

File: .moban.yaml (hunk 7 above)
Comment (SubbulakshmiRS) [reply .moban.yaml]: I tried doing so but this error came up,
     fatal: pathspec 'Description' did not match any files
I was only able to execute the above command when i created a empty file of same name .

File: .ci/generate_bear_requirements.py (hunk 8 above)
Comment (jayvdb) [reply .ci/generate_bear_requirements.py]: @anctartica has a fix for that at https://github.com/coala/coala-bears/pull/2625
We can merge that first, then this one

File: .moban.dt/Description.jj2 [hunk 11]
```diff
@@ -0,0 +1,16 @@
+Package: {{ name }}
+Version: {{ build_version }}
+Title: R dependencies of Coala package
```
Comment (jayvdb) [inline .moban.dt/Description.jj2]: This should be `R dependencies for {{ description }}`

File: .ci/generate_bear_requirements.py [hunk 12]
```diff
@@ -30,6 +30,8 @@
 from dependency_management.requirements.GemRequirement import GemRequirement
 from dependency_management.requirements.NpmRequirement import NpmRequirement
 from dependency_management.requirements.PipRequirement import PipRequirement
+from dependency_management.requirements.RscriptRequirement import RscriptRequirement
+
```
Comment (anctartica) [inline .ci/generate_bear_requirements.py]: remove this blank line

**Comment by anctartica** [comment]
Can you please rebase?

**Comment by jayvdb** [comment]
Zero changes since last time.
Taking too long.
Reassigning issue.

**Pull request closed without merging**